### DISCUSSION TITLE
Will mold add support for "-rpath-link" and DT_RPATH?

### DISCUSSION THREAD

**Discussion by pawelkrysiak** [rui314/mold#365] [Q&A]
Hello,

Currently `-rpath-link` option is ignored by mold.
Also `DT_RPATH` is only mentioned in source code, but not populated (mold uses `DT_RUNPATH` instead).

Are there any plans to support above in future mold releases?
In case we are linking with .so which recursively needs another .so, gold is able to resolve all dependencies while mold can only resolve the direct ones, which causes application to crash at runtime.

**Comment by rui314** [comment]
I'm happy to support it if doing it makes sense, but it looks like LLVM lld can live without supporting it. So I thought that that's deprecated. What project do you need that flag?

**Reply by pawelkrysiak** [reply]
Thanks for being open to supporting even deprecated features.
I've just noticed that someone else requested the same in #420 and that you've already resolved that issue.
So we can treat this as obsolete/closed as well.
